### DISCUSSION TITLE
How to hook into test tear down event?

### DISCUSSION THREAD

**Discussion by lehmannk** [fzyzcjy/flutter_convenient_test#358] [Q&A]
I want to execute some custom reporting after all convenience tests have finished. How can I hook into the teardown method in order to achieve this? I have already tried to add `async` and `await` statements but could not manage to wait for the execution manually. Sorry for the newbiew question....

**Comment by fzyzcjy** [comment]
Hmm, What about `tearDownAll`, a function in Flutter test library?

**Reply by lehmannk** [reply]
Well, this is only applicable after a single group() has run. I need a callback after ALL tests (e.g. multiple groups()) have run. Just before the app exists. And, is there a way to get the report file path from convenience_test after all tests have been reported?

**Reply by fzyzcjy** [reply]
Well, try to put tearDownAll before any groups?

> And, is there a way to get the report file path from convenience_test after all tests have been reported?

Feel free to PR to add one!
